php-frontend/src/test/java/org/sonar/php/cfg/ControlFlowGraphTest.java

```diff
@@ -51,6 +52,141 @@
 
+  @Test
+  public void break_without_argument() {
+    verifyBlockCfg("" +
+      "while (cond( succ = [body, END], elem = 1 )) {" +
+      "  body( succ = [END], elem = 2 );" +
+      "  break;" +
+      "}");
+
+    verifyBlockCfg("" +
+      "do {" +
+      "  body( succ = [END] );" +
+      "  break;" +
+      "} while (cond( succ = [body, END] ));");
+
+    verifyBlockCfg("" +
+      "while (outerCond( succ = [innerCond, END] )) {" +
+      "  while (innerCond( succ = [bodyInner, ifCond] )) {" +
+      "    bodyInner( succ = [innerCond] );" +
+      "  }" +
+      "  if (ifCond( succ = [bodyIf, outerCond] )) {" +
+      "    bodyIf( succ = [END] );" +
+      "    break;" +
+      "  }" +
+      "}");
+  }
+
+  @Test
+  public void continue_without_argument() {
+    verifyBlockCfg("" +
+      "while (cond( succ = [body, END], elem = 1 )) {" +
+      "  body( succ = [cond], elem = 2 );" +
+      "  continue;" +
+      "  dead( succ = [cond], elem = 1);" +
+      "}");
+
+    verifyBlockCfg("" +
+      "do {" +
+      "  body( succ = [body] );" +
+      "  continue;" +
+      "} while (cond( succ = [body, END] ));");
+  }
+
+  @Test
+  public void break_with_argument() {
+    String breakInnerLoop = "" +
+      "while (outerCond( succ = [innerCond, END] )) {" +
+      "  while (innerCond( succ = [ifCond, outerCond] )) {" +
+      "    if (ifCond( succ = [body, innerCond] )) {" +
+      "      body( succ = [outerCond] );" +
+      "      break %s;" +
+      "    }" +
+      "  }" +
+      "}";
+
+    verifyBlockCfg(String.format(breakInnerLoop, "0"));
+    verifyBlockCfg(String.format(breakInnerLoop, "1"));
+
+    verifyBlockCfg("" +
+      "while (outerCond( succ = [innerCond, END] )) {" +
+      "  while (innerCond( succ = [ifCond, outerCond] )) {" +
+      "    if (ifCond( succ = [body, innerCond] )) {" +
+      "      body( succ = [END] );" +
+      "      break 2;" +
+      "    }" +
+      "  }" +
+      "}");
+  }
+
+  @Test
+  public void continue_with_argument() {
+    String continueInnerLoop = "" +
+      "while (outerCond( succ = [innerCond, END] )) {" +
+      "  while (innerCond( succ = [ifCond, outerCond] )) {" +
+      "    if (ifCond( succ = [body, innerCond] )) {" +
+      "      body( succ = [innerCond] );" +
+      "      continue %s;" +
+      "    }" +
+      "  }" +
+      "}";
+
+    verifyBlockCfg(String.format(continueInnerLoop, "0"));
+    verifyBlockCfg(String.format(continueInnerLoop, "1"));
+
+    verifyBlockCfg("" +
+      "while (outerCond( succ = [innerCond, END] )) {" +
+      "  while (innerCond( succ = [ifCond, outerCond] )) {" +
+      "    if (ifCond( succ = [body, innerCond] )) {" +
+      "      body( succ = [outerCond] );" +
+      "      continue 2;" +
+      "    }" +
+      "  }" +
+      "}");
+
+    verifyBlockCfg("" +
+      "do {" +
+      "  while (innerCond( succ = [ifCond, outerCond] )) {" +
+      "    if (ifCond( succ = [body, innerCond] )) {" +
+      "      body( succ = [innerCond] );" +
```

Comment:
shouldn't this be `outerCond`?

Reply:
you are right! I thought that `continue` continues to start of a loop